pinot-core/src/test/java/org/apache/pinot/queries/JsonMatchQueriesTest.java

```diff
@@ -0,0 +1,199 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.queries;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.commons.io.FileUtils;
+import org.apache.pinot.common.response.broker.BrokerResponseNative;
+import org.apache.pinot.segment.local.indexsegment.immutable.ImmutableSegmentLoader;
+import org.apache.pinot.segment.local.segment.creator.impl.SegmentIndexCreationDriverImpl;
+import org.apache.pinot.segment.local.segment.index.loader.IndexLoadingConfig;
+import org.apache.pinot.segment.local.segment.readers.GenericRowRecordReader;
+import org.apache.pinot.segment.spi.ImmutableSegment;
+import org.apache.pinot.segment.spi.IndexSegment;
+import org.apache.pinot.segment.spi.creator.SegmentGeneratorConfig;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.config.table.TableType;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.apache.pinot.spi.data.Schema;
+import org.apache.pinot.spi.data.readers.GenericRow;
+import org.apache.pinot.spi.utils.builder.TableConfigBuilder;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+
+/**
+ * Queries test for JSON_MATCH predicate.
+ */
+public class JsonMatchQueriesTest extends BaseQueriesTest {
```

Comment:
Can we rename this test file to `JsonMatchV2QueriesTest` and rename the existing `JsonMatchPredicateTest` to `JsonMatchV1QueriesTest` so that the scope of these tests is clear with respect to different versions of JsonMatch?

Reply:
The current `JsonMatchPredicateTest` has several tests on `json_extract_scalar`, so I think it might be better to rename it to `JsonQueriesTest`?
I'll add some V1 format queries into the new `JsonMatchQueriesTest` to ensure the backward compatibility

Reply:
All the queries from `JsonMatchPredicateTest` got included in `JsonDataTypeTest`, so the sole purpose of JsonMatchPredicateTest now is to ensure that JSON_MATCH V1 doesn't regress. Hence, my suggestion to rename it to JsonMatchV1QueriesTest and after that we don't need to maintain this test file. I will modify `JsonDataTypeTest` to use V2 JSON_MATCH and add a few more tests and it will continue to test json_extract_scalar with JSON data type. While we deprecate JsonMatchV1QueriesTest along with JSON_MATCH V1, the new test cases will be able to validate all the new json functionality.

Reply:
In that case, I would recommend directly modifying the `JsonMatchPredicateTest` to the new `JsonDataTypeTest`. We don't need to keep it because I already add the legacy queries test into the `JsonMatchPredicateTest`, so both versions query are covered in one test.